File: contract/scenarios/SIM-10-bypass-fixture.ts
// SIM-10 bypass fixture — DO NOT USE IN PRODUCTION
//
// This file intentionally imports the v1 SDK and calls evaluate() directly,
// bypassing the Enforce wrapper. It exists solely so the enforce-no-bypass
// lint can prove it catches this pattern (SIM-10).
//
// The lint should reject this file with an enforce-no-bypass violation.

import { AtlasentClient } from "@atlasent/sdk";

const client = new AtlasentClient({
  apiKey: "test_key",
  baseUrl: "https://api.atlasent.io",
});

// VIOLATION: calling evaluate() directly instead of going through Enforce.run()
export async function bypassEnforce(action: string, agent: string) {
  const result = await client.evaluate({ action, agent });
  if (result.decision === "allow") {
    // Executing without verifyPermit — this is exactly what Enforce prevents.
    return { ok: true };
  }
  return { ok: false };
}

File: contract/scenarios/SIM-10-bypass-fixture.py
# SIM-10 bypass fixture — DO NOT USE IN PRODUCTION
#
# This file intentionally imports the v1 SDK and calls evaluate() directly,
# bypassing the Enforce wrapper. It exists solely so the enforce-no-bypass
# lint can prove it catches this pattern (SIM-10).
#
# The lint should reject this file with an enforce-no-bypass violation.

from atlasent import AtlasentClient


async def bypass_enforce(action: str, agent: str) -> dict:
    client = AtlasentClient(api_key="test_key", base_url="https://api.atlasent.io")

    # VIOLATION: calling evaluate() directly instead of going through Enforce.run()
    result = await client.evaluate({"action": action, "agent": agent})
    if result.decision == "allow":
        # Executing without verify_permit — this is exactly what Enforce prevents.
        return {"ok": True}
    return {"ok": False}
